docs/src/tutorials/point_process_simulation.md

```diff
@@ -0,0 +1,254 @@
+
+In JumpProcesses a `VariableRateJump` is a TPP whose rate is allowed to vary at
+arbitrary times. Again, we start by declaring the rate function and the affect.
+
+```@example tpp-tutorial
+seasonal_rate(u, p, t) = 1 + (u[1] % 2 == 0 ? sin(t) : cos(t))
+seasonal_affect!(integrator) = (integrator.u[2] += 1)
+```
+
+There are algorithms for simulating TPPs which can take
+advantage of bounded variable rates. In our example, ``\lambda_2`` is
+bounded above by ``2`` and below by ``1``. To initialize, a bounded
+`VariableRateJump` we must supply the rate upper-bound and the
+interval for which the upper-bound is valid. In this case, the bound is
+valid throughout time. The lower-bound is optional but can improve the
+speed of the simulation.
+
+```@example tpp-tutorial
+urate(u, p, t) = 2           # upper bound
+rateinterval(u, p, t) = Inf  # time window bound is valid over
+lrate(u, p, t) = 1           # lower bound
+```
+
+Now, we can declare the seasonal process as a `VariableRateJump`.
+
+```@example tpp-tutorial
+seasonal_process = VariableRateJump(seasonal_rate, seasonal_affect!;
+    urate, rateinterval, lrate)
+```
+
+We initialize a new base problem with a different simulation
+specification. Since we have a multivariate process, the state of the
+system is a vector with two counts.
+
+```@example tpp-tutorial
+u0 = [0, 0]
+tspan = (0.0, 10.0)
+dprob = DiscreteProblem(u0, tspan)
+```
+
+We also need to modify [`JumpProblem`](@ref) to use a simulation algorithm (i.e.
+aggregator) that supports bounded `VariableRateJump`s. In this case, we use the
+`Coevolve` method, which is another type of thinning algorithm for multivariate
+process and which is an improvement of the Ogata thinning method. This method
+also requires a dependency graph that indicates for a given TPP which other TPPs
+have rates that depend on states/variables altered in its affect function. Note
+JumpProcesses' convention is that a given TPP should also always be a dependency
+of itself. Internally, JumpProcesses preserves the relative ordering of point
+processes of each distinct type, but always reorders all `ConstantRateJump`s to
+appear before any `VariableRateJump`s. Irrespective of how `JumpProblem` is
+initialized, internally the processes will be ordered as the vector
+`[poisson_process, seasonal_process]` so that these will have the internal
+indexes of `1` and `2` respectively. Note, this vector of the processes is
+distinct from our state variable vector, `u`. When `poisson_process` fires
+`u[1]` is altered, and as the rate for `seasonal_process` depends on it we have
+that the dependencies of `poisson_process` are `[1,2]`. In contrast, the rate of
+`poisson_process` is independent of `u[2]` which `seasonal_process` modifies,
+and hence the dependencies of `seasonal_process` are only `[2]`.
+
+Therefore, we obtain the following dependency graph:
+
+```@example tpp-tutorial
+dep_graph = [[1,2], [2]]
```

Comment:
@gzagatti I changed this as it seemed reversed in your PR. Can you double check this?

Reply:
That was a mistake. Your fix is correct.